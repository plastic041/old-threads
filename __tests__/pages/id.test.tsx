import { render } from "@testing-library/react";
import mockRouter from "next-router-mock";
import IdPage from "~/pages/threads/[id]";

jest.mock("next/router", () => require("next-router-mock"));
jest.mock("next/dist/client/router", () => require("next-router-mock"));

describe("IdPage", () => {
  beforeEach(() => {
    mockRouter.setCurrentUrl("/threads/616");
  });

  it("renders correctly", async () => {
    const view = render(<IdPage />);
    expect(view).toMatchSnapshot();
  });
});
