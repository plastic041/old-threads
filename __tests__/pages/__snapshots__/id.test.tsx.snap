// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdPage renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="flex flex-row items-center gap-4 bg-white px-4 py-2"
      >
        <nav>
          <a
            class="text-teal-700 hover:text-teal-500"
            href="/"
          >
            <svg
              class="h-8 w-8"
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M6.85355 3.14645C7.04882 3.34171 7.04882 3.65829 6.85355 3.85355L3.70711 7H12.5C12.7761 7 13 7.22386 13 7.5C13 7.77614 12.7761 8 12.5 8H3.70711L6.85355 11.1464C7.04882 11.3417 7.04882 11.6583 6.85355 11.8536C6.65829 12.0488 6.34171 12.0488 6.14645 11.8536L2.14645 7.85355C1.95118 7.65829 1.95118 7.34171 2.14645 7.14645L6.14645 3.14645C6.34171 2.95118 6.65829 2.95118 6.85355 3.14645Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </nav>
        <h1
          class="text-[16px] md:text-[32px]"
        >
          Loading...
        </h1>
      </header>
      <div
        class="mt-40 flex w-full flex-grow items-center justify-center"
      >
        <span
          class="animate-pulse text-teal-900"
        >
          페이지를 불러오는 중이에요 \\(￣︶￣*\\))
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="flex flex-row items-center gap-4 bg-white px-4 py-2"
    >
      <nav>
        <a
          class="text-teal-700 hover:text-teal-500"
          href="/"
        >
          <svg
            class="h-8 w-8"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M6.85355 3.14645C7.04882 3.34171 7.04882 3.65829 6.85355 3.85355L3.70711 7H12.5C12.7761 7 13 7.22386 13 7.5C13 7.77614 12.7761 8 12.5 8H3.70711L6.85355 11.1464C7.04882 11.3417 7.04882 11.6583 6.85355 11.8536C6.65829 12.0488 6.34171 12.0488 6.14645 11.8536L2.14645 7.85355C1.95118 7.65829 1.95118 7.34171 2.14645 7.14645L6.14645 3.14645C6.34171 2.95118 6.65829 2.95118 6.85355 3.14645Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </nav>
      <h1
        class="text-[16px] md:text-[32px]"
      >
        Loading...
      </h1>
    </header>
    <div
      class="mt-40 flex w-full flex-grow items-center justify-center"
    >
      <span
        class="animate-pulse text-teal-900"
      >
        페이지를 불러오는 중이에요 \\(￣︶￣*\\))
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
